_posts/2024-03-11-release-0_5_0.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title:  "Kroxylicious release 0.5.0"
+date:   2024-03-10 00:00:00 +0000
+author: "Robert Young"
+author_url: "https://www.github.com/robobario"
+categories:  [releases, kroxylicious, record-encryption]
+---
+
+The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.5.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0).
+
+### Record Encryption
+
+This release was focused on refining [Record Encryption](/use-cases) to be ready for experimentation in secure environments, adding:
+* Enable users to configure Hashicorp Vault TLS, providing custom keystore and truststore
+* Enable users to supply the Hashicorp Vault token via file rather than having to inject it into the proxy configuration
+* Include Record Encryption filter (and other Kroxylicious project supported filters) in the binary distribution
+* Release a Docker image `quay.io/kroxylicious/kroxylicious` also containing the supported filters
+* Internal refactoring to control the usage of key material, preparing to support alternate Ciphers and configurable Additional Authenticated Data
+
+The protocol for immutable encrypted data written to the broker is now aligned with our initial design and we guarantee data encrypted with version 0.5.0 
+of the Filter will be decryptable forever (assuming the keys stored in the KMS are available forever for decryption).
+
+> **_NOTE:_**: Until now we have used the term Envelope Encryption when describing the Filter. Going forward we will use the name Record Encryption 
+to better describe the granularity that encryption is happening at. Encryption is done per [record](https://kafka.apache.org/documentation/#record).
+Envelope Encryption is how we encrypt those records. Record Encryption is how we implement Encryption-At-Rest.
+
+### Other Improvements
+
+* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is forwards-compatible, but must be modified if we wish to intercept new
+APIs or fields added to existing APIs)
+* A fix for upstream TLS connections failing, [@callaertanthony](https://github.com/callaertanthony)'s first PR contribution, thank you!
+* Added a `lowestTargetBrokerId` configuration for Port-per-broker Virtual Cluster exposition
+* Numerous bugfixes, documentation improvements, test enhancements
+
+### Artefacts
+
+You'll find binaries attached to the GitHub [release](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0), and available through [Maven Central](https://repo1.maven.org/maven2/io/kroxylicious/kroxylicious-app/0.5.0/).
+
+Docker images are hosted at [quay.io/kroxylicious/kroxylicious](https://quay.io/repository/kroxylicious/kroxylicious)
```

Comment:
Might be nice to add a line or two at the end here about where people can report bugs and leave feedback